parsec/core/fs/fs.py

```diff
@@ -270,15 +270,14 @@ def _iter_workspaces(self):
         read_right: bool = True,
         write_right: bool = True,
     ) -> None:
-        cooked_path = FsPath(path)
-        if not cooked_path.is_workspace():
+        workspace_name, subpath = self._split_path(path)
+        if not subpath:
             # Will fail with correct exception
             workspace_name = path
-        else:
-            workspace_name = cooked_path.workspace
+        workspace_entry = self._get_workspace_entry_from_name(workspace_name)
 
         await self._user_fs.workspace_share(
-            workspace_name,
+            workspace_entry.access.id,
```

Comment:
How could the tests miss that? :thinking:

Reply:
The tests are already ported to `UserFS` api... :thinking: